src/libsync/propagatedownload.cpp

```diff
@@ -344,17 +359,30 @@ void PropagateDownloadFile::start()
     qCDebug(lcPropagateDownload) << _item->_file << propagator()->_activeJobList.count();
     _stopwatch.start();
 
+    auto &syncOptions = propagator()->syncOptions();
+    auto vfs = syncOptions._vfs;
+
     // For virtual files just create the file and be done
+    if (_item->_type == ItemTypeVirtualFileDehydration) {
+        _item->_type = ItemTypeVirtualFile;
+        // TODO: Could dehydrate without wiping the file entirely
+        auto fn = propagator()->getFilePath(_item->_file);
+        qCDebug(lcPropagateDownload) << "dehydration: wiping base file" << fn;
+        QFile::remove(fn);
+        propagator()->_journal->deleteFileRecord(_item->_file);
```

Comment:
I'd delete the file record before deleting the file, so then if the client crashes here, the file won't be removed on the server.

Actually, the test `test/testdatabaseerror.cpp`  could test that if we make it test virtual files (which it currently does not test)